spec/index.html

```diff
@@ -8941,13 +8953,120 @@ <h5>Collect <code>FILTER</code> Elements</h5>
+                  <p>
+                    One way to provide `BindingInScope` is to change `eval` to also have the current row
+                    as an argument or a "null" token. . Normally this the "null" token. This would be set in
+                    eval-filter.
+                  </p>
+                  <p>
+                    Another way is to have a global (to the execution) variable. In eval-exists, the
+                    old value is recorded, the global set to the new value, and reset on exit - this forms a
+                    stack for nested EXISTS.
+                  </p>
+                  <p>
+                    Modifing `eval` is better prepartion for other correlted operations.<br/>
+                    A current-row stack in EXISTS isless intrusive.
+                  </p>
+                </div>
+                
+                <div class="defn">
+                  <b>Definition: <span id="defn_projmap">Projection Expression Variable Remapping</span></b>
+                  <p>
+                    For a projection algebra operation #sparqlProjection `Project(A, PV)` acting on algreg express `A` and with set of variables `PV`, define
+                    a partial mapping `F` from
+                    `<a href="#sparqlQueryVariables">V</a>`,
+                    the set of all variables, to `V` where:
+                  </p>
+                  <pre>F(v) = v1 if v is in PV, where v1 is a fresh variable
+F(v) = v if v is not in PV</pre>
+                  <p>
+                    Define the Projection Expression Variable Remapping `ProjectMap(P, PV)`
+                  </p>
+                  <pre>ProjectMap(Project(A, PV)) = Project(A1, PV) 
+                 where A1 is the result of applying F
+                 to every variable mentioned in A.</pre>
+                  <p>
+                    The Projection Expression Variable Remapping yields an algrebra expression that
+                    evaluates to the same results as the Project argument. No variable of `ProjectMap(Project(A, PV))`
+                    that is not in `PV` is mentioned anywhere else in the algebra expression for the query.
+                  </p>
+                </div>
+                <p>This process is applied throughout the graph pattern of <code>EXISTS</code>:</p>
+                <div class="defn">
+                  <b>Definition: <span id="defn_varrename">Variable Remapping</span></b>
+                  <p>
+                    For any algebra expression `X`, define the Variable Remapping `PrjMap(X)` 
+                    of algebra expression `X`:
+                  </p>
+                  <pre>PrjMap(X) = replace all project operations Project(P, PV) 
+      with ProjectMap(P, PV) for each projection in X.</pre>
+                </div>
+                <p>
+                  The outcome of `PrjMap` is independent of the order of replacement
+                  (e.g. bottom-up or top-down).
```

Comment:
```suggestion
                  (e.g., bottom-up or top-down).
```

Reply:
or, because `bottom-up` is often misunderstood
```suggestion
                  (e.g., inside-out, or outside-in).
```

Reply:
or even
```suggestion
                  (e.g., bottom-up, top-down, inside-out, or outside-in).
```